File: scripts/package/disease_metadata_omim.py
from __future__ import annotations

import gc
from pathlib import Path

from utils.log_util import get_logger
from package.rdf_build_support import (
    load_config
)
from package.disease_metadata_util import (
    load_omim_disease_ids,
    load_shared_reference_data,
    write_omim_disease_ttl,
)

logger = get_logger()


def disease_metadata_omim() -> None:
    logger.info("start OMIM disease metadata RDF build")
    config = load_config('config.ini')
    omim_ids = None
    reference_data = None

    try:
        omim_ids = load_omim_disease_ids(config['omim_mim2gene_data_path'])
        logger.info("OMIM disease count: %s", len(omim_ids))

        reference_data = load_shared_reference_data(
            config['medgen_omim_hpo_path'],
            config['mondo_owl_path'],
            config['kegg_disease_path'],
            config['genereviews_omim_path']
        )
        logger.info("OMIM inheritance count: %s", len(reference_data.inheritance_map))

        append_unique(omim_ids, reference_data.mappings.omim_to_mondo.keys())
        logger.info("OMIM KEGG count: %s", len(reference_data.kegg_map))
        logger.info("OMIM GeneReviews count: %s", len(reference_data.gene_reviews_map))

        output_path = Path(config['rdf_output_dir']) / "OMIM.ttl"
        write_omim_disease_ttl(
            output_path,
            omim_ids,
            reference_data.inheritance_map,
            reference_data.mappings,
            reference_data.kegg_map,
            reference_data.gene_reviews_map,
        )
        logger.info("finished OMIM disease metadata RDF build: output=%s diseases=%s", output_path, len(omim_ids))
    finally:
        reference_data = None
        if omim_ids is not None:
            omim_ids.clear()
        gc.collect()


def append_unique(values: list[str], additions) -> None:
    before_count = len(values)
    seen = set(values)
    for value in additions:
        if value in seen:
            continue
        seen.add(value)
        values.append(value)
    logger.info("appended unique values: before=%s after=%s added=%s", before_count, len(values), len(values) - before_count)


if __name__ == "__main__":
    disease_metadata_omim()

File: scripts/java/DiseaseMetadataOMIM.java
package pubcasefinder_260415;

import java.util.LinkedHashMap;
import java.util.LinkedHashSet;

// OMIM 질환 메타데이터를 조합해 OMIM.ttl을 생성한다.
public class DiseaseMetadataOMIM {
	/** OMIM ID 집합과 외부 매핑 정보를 결합해 OMIM.ttl을 생성한다. */
	public static void main(String[] args) throws Exception {
		// 명시적인 OMIM 질환 ID 집합을 시작점으로 삼고, 외부 매핑과 참조 정보를 덧붙인다.
		LinkedHashSet<String> omimIds = DiseaseMetadataUtil.loadOmimDiseaseIds(DiseaseMetadataUtil.OMIM_MIM2GENE_PATH);
		System.out.println("OMIM All Count : " + omimIds.size());

		DiseaseMetadataUtil.SharedReferenceData referenceData =
				DiseaseMetadataUtil.loadSharedReferenceData();
		System.out.println("OMIM inheritance Count : " + referenceData.inheritanceMap.size());
		omimIds.addAll(referenceData.mappings.omimToMondo.keySet());
		System.out.println("OMIM KEGG Count : " + referenceData.keggMap.size());
		System.out.println("OMIM Gene_Review Count : " + referenceData.geneReviewsMap.size());

		DiseaseMetadataUtil.writeOmimDiseaseTtl(
				DiseaseMetadataUtil.RDF_DIR + "/OMIM.ttl",
				omimIds,
				referenceData.inheritanceMap,
				referenceData.mappings,
				referenceData.keggMap,
				referenceData.geneReviewsMap);
		System.out.println("OMIM All Count : " + omimIds.size());
	}
}
